File: 1364-tuple-with-same-product/1364-tuple-with-same-product.c
int tupleSameProduct (int* nums, int numsSize) {
    int *multiple = calloc (100000001, sizeof (int));
    int product = 0, result = 0;
    for (int i = 0; i < numsSize; i++) {
        for (int j = 0; j < i; j++) {
            product = nums [i] * nums [j];
            result += (multiple [product] * 8);
            multiple [product]++;
        }
    }
    free (multiple);
    return result;
}

File: 1364-tuple-with-same-product/1364-tuple-with-same-product.java
class Solution {
    public int tupleSameProduct (int [] nums) {
        Map <Integer, Integer> productCount = new HashMap <> ();
        int n = nums.length;
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                int product = nums [i] * nums [j];
                productCount.put (product, productCount.getOrDefault (product, 0) + 1);
            }
        }
        int total = 0;
        for (int count : productCount.values ()) {
            if (count >= 2) {
                total += count * (count - 1) / 2;
            }
        }
        return total * 8;
    }
}